hudi-flink/src/test/java/org/apache/hudi/sink/TestWriteCopyOnWrite.java

```diff
@@ -380,12 +380,12 @@ public void testUpsertWithDelete() throws Exception {
   @Test
   public void testInsertWithMiniBatches() throws Exception {
     // reset the config option
-    conf.setDouble(FlinkOptions.WRITE_BATCH_SIZE, 0.0006); // 630 bytes batch size
+    conf.setDouble(FlinkOptions.WRITE_BATCH_SIZE, 0.00075); // 786 bytes batch size
```

Comment:
are these the tests you were mentioning were failing in CI? ideally these tests should not be impacted by this patch right?

Reply:
Yes. I ran the tests from the master branch on my local and they seem to fail for these as well. What I could figure figure out was the size estimate of the HoodieRecord calculated by HoodieRecordSizeEstimator was higher from what was being assumed in the testcase. Hence the batch size was being set to a lower value in the testcase causing the flush to happen earlier than expected. I have modified the batch size based on the output of HoodieRecordSizeEstimator. Not sure if this is the right fix though.

Reply:
hmmm, strange. I see quite a few PRs shows green. so unlikely this fails consistently. also, if there are any such unrelated failures, lets isolate them to separate PR. I will try to reproduce the issue locally in master as well.

Reply:
This does seem unrelated. @nsivabalan Please try to reproduce this before landing this.

Reply:
@danny0405 @leesf @yanghua : hey folks. Can one of you check why these tests are failing for this patch and why the fix is required. I also tried locally and it is failing w/ this patch, but don't have lot of context around tests. I expected these tests should not be affected by the changes in this patch.

Reply:
Also, nishith left a comment else where about backwards compatability. Can you check that out as well.

Reply:
Thanks for the reminder, the `TestWriteCopyOnWrite#testInsertWithMiniBatches` actually test flushing data buffer based on the buffer size, the `StreamWriteFunction` detect the record buffer size using Java Reflection, does this patch affect the record size (for e.g, the data structure of the record) ?

Reply:
Is it related with the JDK version ? What JDK version did you use @vaibhav-sinha ?

Reply:
I used jdk1.8.0_171.jdk.

Reply:
I think the reason for this is as follows:

For timestamp field, HoodieAvroUtils now returns an instance of LocalDateTime instead of Long. When Row data is being converted to HoodieRecordPayload, then `orderingVal` is being fetched using `HoodieAvroUtils.getNestedFieldVal`. Hence the structure of the payload has changed and is of larger size than before. 

```
public HoodieRecordPayload<?> createPayload(GenericRecord record, boolean isDelete) throws Exception {
    if (shouldCombine) {
      ValidationUtils.checkState(preCombineField != null);
      Comparable<?> orderingVal = (Comparable<?>) HoodieAvroUtils.getNestedFieldVal(record,
          preCombineField, false);
      return (HoodieRecordPayload<?>) constructor.newInstance(
          isDelete ? null : record, orderingVal);
    } else {
      return (HoodieRecordPayload<?>) this.constructor.newInstance(Option.of(record));
    }
  }
```

Reply:
Okey, got your idea, can we keep the type as long if the data type is timestamp without time zone (or UTC timezone), we can keep the compatibility and the long has less memory footprint as you have tested.

This is important because Hoodie writer handle buffer the records, so the less memory the better.